src/utils/error.ts

```diff
@@ -0,0 +1,55 @@
+import { captureException } from '@sentry/react'
+import { AxiosError } from 'axios'
+
+import { BFF_API } from 'constants/env'
+import { ROUTES_API_PATHS } from 'constants/index'
+
+const ErrorInfo = {
+  routeApiError: 0,
+  iamApoError: 0,
+  errorThreshold: 2,
+  sentAlertIamApi: false,
+  sentAlertRouteApi: false,
+}
+
+const isIamApiDown = () => ErrorInfo.iamApoError > ErrorInfo.errorThreshold
+export const isRouteApiDown = () => ErrorInfo.routeApiError > ErrorInfo.errorThreshold
+
+const sendError = (name: string, apiUrl: string) => {
+  const error = new Error(`${name} Error: ${apiUrl}`)
+  error.name = `${name} was down`
+  captureException(error, { level: 'fatal', extra: { args: JSON.stringify({ apiUrl }, null, 2) } })
+}
+
+/**
+ * check error status: blocked, maybe cors issues or  server down
+ * only check bff api + 2 route apis
+ */
+export const checkIamDown = (axiosErr: AxiosError) => {
+  const isDie =
+    navigator.onLine &&
+    (!axiosErr?.response?.data ||
+      (axiosErr?.response?.status === 404 && axiosErr?.response?.data === '404 page not found'))
```

Comment:
Will it every return 503 and still have data?

Reply:
<img width="772" alt="image" src="https://github.com/KyberNetwork/kyberswap-interface/assets/33005392/40f733f9-be7d-43fe-b2c1-bbc1cedc9a23">
<img width="734" alt="image" src="https://github.com/KyberNetwork/kyberswap-interface/assets/33005392/d73b50b4-4a16-4efb-a670-95bd0aafb61b">

No.
checked 503, 529